### PR TITLE
Upgrade highway to 1.2.0

### PR DESCRIPTION
**Pull request by mtl1979**

### CODE REVIEW

**Comment by Begasus** [comment]
As mentioned on IRC, this could be turned into using shared libraries, saved fiddling with static ones.
Also add a devel package and use libVersion(Compat) where needed. :)

File: dev-libs/highway/highway-1.2.0.recipe [hunk 1]
```diff
@@ -13,23 +13,33 @@ Why Highway?
...
 SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	highway$secondaryArchSuffix = $portVersion
-	devel:libhwy$secondaryArchSuffix = $portVersion
-	devel:libhwy_contrib$secondaryArchSuffix = $portVersion
+	highway$secondaryArchSuffix = $portVersion compat >= 1
+	lib:libhwy$secondaryArchSuffix = $portVersion
+	lib:libhwy_contrib$secondaryArchSuffix = $portVersion
+	lib:libhwy_test$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline dev-libs/highway/highway-1.2.0.recipe:28]: Use libVersionCompat here (libVersion = 1.2.0 compat = 1) and for all the libraries in this package (including _devel)

File: dev-libs/highway/highway-1.2.0.recipe [hunk 2]
```diff
@@ -13,23 +13,33 @@ Why Highway?
...
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
 	"
 
+PROVIDES_devel="
+	devel:libhwy$secondaryArchSuffix = $portVersion
+	devel:libhwy_contrib$secondaryArchSuffix = $portVersion
+    devel:libhwy_test$secondaryArchSuffix = $portVersion
+    "
```
Comment (Begasus) [inline dev-libs/highway/highway-1.2.0.recipe:38]: Use 1 tab, no spaces

File: dev-libs/highway/highway-1.2.0.recipe [hunk 3]
```diff
@@ -52,9 +63,10 @@ INSTALL()
 	make -C build install
 
 	prepareInstalledDevelLibs \
-		libhwy libhwy_contrib
+		libhwy libhwy_contrib libhwy_test
```
Comment (Begasus) [inline dev-libs/highway/highway-1.2.0.recipe:66]: Nitpicking :) You could move these libraries into separate lines

File: dev-libs/highway/highway-1.2.0.recipe [hunk 4]
```diff
@@ -13,23 +13,35 @@ Why Highway?
...
-CHECKSUM_SHA256="e1d47ce510429fdcf31f41697ca74fb0dcd59d933196e531a86d51751a56f4cc"
+REVISION="1"
+SOURCE_URI="https://github.com/google/highway/releases/download/$portVersion/highway-$portVersion.tar.gz"
+CHECKSUM_SHA256="58e9d5d41d6573ad15245ad76aec53a69499ca7480c092d899c4424812ed906f"
 SOURCE_FILENAME="highway-$portVersion.tar.gz"
 
 ARCHITECTURES="all !x86_gcc2"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersionCompat="$portVersion compat >= ${portVersion%%.*}"
```
Comment (Begasus) [inline dev-libs/highway/highway-1.2.0.recipe:24]: Format: https://github.com/Begasus/haikuports/wiki/Useful-variables-for-recipes#libversion-and-libversioncompat

Comment (Begasus) [reply dev-libs/highway/highway-1.2.0.recipe:24]: If debugInfoPackage is involved you would need $libVersion along with $libVersionCompat.

File: dev-libs/highway/highway-1.2.0.recipe (hunk 3 above)
Comment (mtl1979) [reply dev-libs/highway/highway-1.2.0.recipe:68]: I know... Usually it's done if the line gets really long... But it also helps if the recipe gets constantly edited... However it doesn't allow appending to the list without changing at least one extra line.

File: dev-libs/highway/highway-1.2.0.recipe [hunk 5]
```diff
@@ -13,23 +13,36 @@ Why Highway?
...
-CHECKSUM_SHA256="e1d47ce510429fdcf31f41697ca74fb0dcd59d933196e531a86d51751a56f4cc"
+REVISION="1"
+SOURCE_URI="https://github.com/google/highway/releases/download/$portVersion/highway-$portVersion.tar.gz"
+CHECKSUM_SHA256="58e9d5d41d6573ad15245ad76aec53a69499ca7480c092d899c4424812ed906f"
 SOURCE_FILENAME="highway-$portVersion.tar.gz"
 
 ARCHITECTURES="all !x86_gcc2"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersion="1.2.0"
```
Comment (Begasus) [inline dev-libs/highway/highway-1.2.0.recipe:24]: In this case libVersion is the same as portVersion, reduces git noise on a next update.

Comment (mtl1979) [reply dev-libs/highway/highway-1.2.0.recipe:24]: So is `libVersion=$portVersion` better?

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks!

**Pull request merged**